portfolio/src/main/webapp/script.js

```diff
@@ -50,17 +64,65 @@ function setTabEvents(tabs) {
         });
 
         tab.addEventListener("click", function() {
+            if (prevSelectedTab().getAttribute("title") === galleryTitle) {
+                toggleGallerySelection(false);
+            }
+            else if (this.getAttribute("title") === galleryTitle) {
+                toggleGallerySelection(true);
+            }
             switchTabSelection(prevSelectedTab(), this);
         });
     }
 } // setTabEvents
 
+
+// GALLERY FEATURES
+
+function getCaption(galleryPic) {
+    return galleryPic.childNodes[1];
+}
+
+function fadeAnimation(caption, fadeIn) {
```

Comment:
The animation looks great! Like the tab hovers, it can also be done purely with CSS. There is no right or wrong way. If you'd like to try the CSS method, a hint is that you can use pseudo classes on parents in CSS selectors. Ex:
.picture:hover caption {
  /* Show caption */
}

Reply:
That's so much better than triggering the animation with javascript! I'm definitely going to give it a shot, it will save me a lot of trouble in the future.